File: .docs-ops/integration-tests/flutter/preamble.dart
// preamble.dart — Context variables for doc snippet type-checking.
// NOT used directly; the extractor wraps each extracted block in a
// function that has these parameters in scope.
//
// Types verified against .docs-ops/sdk-surface/flutter.json before declaring.
// ITERATION NOTE: Add missing declarations as failures surface.
import 'package:amity_sdk/amity_sdk.dart';

// ignore_for_file: unused_element, unused_local_variable

Future<void> docSnippet({
  // Setup
  required String apiKey,
  required String region,
  required String userId,
  required String displayName,
  required String authToken,
  // Chat context
  required String channelId,
  required String subChannelId,
  required String messageId,
  required AmityChannel channel,
  required AmityMessage message,
  // Social context
  required String postId,
  required String communityId,
  required String commentId,
  required String pollId,
  required AmityPost post,
  required AmityComment comment,
  required AmityCommunity community,
  // User
  required String targetUserId,
  required AmityUser user,
  // File / media
  required String fileId,
}) async {
  // <SNIPPET_BODY> (extractor inserts each block here)
}

File: .docs-ops/integration-tests/android/preamble.kt
// preamble.kt — context for Android/Kotlin doc snippets.
// The extractor wraps each block inside a docSnippet function with these params in scope.
// Types verified against .docs-ops/sdk-surface/android.json.

import com.amity.socialcloud.sdk.api.core.AmityCoreClient
import com.amity.socialcloud.sdk.api.chat.AmityChatClient
import com.amity.socialcloud.sdk.api.social.AmitySocialClient
import com.amity.socialcloud.sdk.model.chat.channel.AmityChannel
import com.amity.socialcloud.sdk.model.chat.message.AmityMessage
import com.amity.socialcloud.sdk.model.social.post.AmityPost
import com.amity.socialcloud.sdk.model.social.community.AmityCommunity
import com.amity.socialcloud.sdk.model.social.comment.AmityComment
import com.amity.socialcloud.sdk.model.core.user.AmityUser
import com.amity.socialcloud.sdk.model.social.poll.AmityPoll

fun docSnippet(
    apiKey: String = "",
    userId: String = "user-id",
    displayName: String = "User",
    authToken: String = "token",
    channelId: String = "channel-id",
    subChannelId: String = "sub-channel-id",
    messageId: String = "message-id",
    postId: String = "post-id",
    communityId: String = "community-id",
    commentId: String = "comment-id",
    pollId: String = "poll-id",
    fileId: String = "file-id",
    targetUserId: String = "target-user-id",
    channel: AmityChannel? = null,
    message: AmityMessage? = null,
    post: AmityPost? = null,
    comment: AmityComment? = null,
    community: AmityCommunity? = null,
    user: AmityUser? = null,
    poll: AmityPoll? = null
) {
    // <SNIPPET_BODY>
}
